src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Extensions/ElementNodeExtension.cs

```diff
@@ -8,94 +8,94 @@ namespace Microsoft.Health.Fhir.Anonymizer.Core.Extensions
         {
             return node != null && string.Equals(node.InstanceType, Constants.BundleTypeName, StringComparison.InvariantCultureIgnoreCase);
         }
 
-        public static bool IsReferenceNode(this ElementNode node)
+        public static bool IsReferenceNode(this ITypedElement node)
         {
             return node != null && string.Equals(node.InstanceType, Constants.ReferenceTypeName, StringComparison.InvariantCultureIgnoreCase);
         }
 
-        public static bool IsPostalCodeNode(this ElementNode node)
+        public static bool IsPostalCodeNode(this ITypedElement node)
         {
             return node != null && string.Equals(node.Name, Constants.PostalCodeNodeName, StringComparison.InvariantCultureIgnoreCase);
         }
 
-        public static bool IsReferenceStringNode(this ElementNode node)
-        {
-            return node != null &&
-                node.Parent.IsReferenceNode() &&
-                string.Equals(node.Name, Constants.ReferenceStringNodeName, StringComparison.InvariantCultureIgnoreCase);
-        }
-
-        public static bool IsEntryNode(this ElementNode node)
+        public static bool IsEntryNode(this ITypedElement node)
         {
             return node != null && string.Equals(node.Name, Constants.EntryNodeName, StringComparison.InvariantCultureIgnoreCase);
         }
 
-        public static bool IsContainedNode(this ElementNode node)
+        public static bool IsContainedNode(this ITypedElement node)
         {
             return node != null && string.Equals(node.Name, Constants.ContainedNodeName, StringComparison.InvariantCultureIgnoreCase);
         }
 
-        public static bool HasContainedNode(this ElementNode node)
+        public static bool HasContainedNode(this ITypedElement node)
         {
             return node != null && node.Children(Constants.ContainedNodeName).Any();
         }
 
-        public static bool IsFhirResource(this ElementNode node)
+        public static bool IsFhirResource(this ITypedElement node)
         {
             return node != null && (node.Definition?.IsResource ?? false);
         }
 
-        public static string GetFhirPath(this ElementNode node)
+        public static string GetFhirPath(this ITypedElement node)
         {
             return node == null ? string.Empty : Regex.Replace(node.Location, s_locationToFhirPathRegex, string.Empty);
         }
 
-        public static string GetNodeId(this ElementNode node)
+        public static string GetNodeId(this ITypedElement node)
         {
             var id = node.Children("id").FirstOrDefault();
             return id?.Value?.ToString() ?? string.Empty;
         }
 
-        public static ElementNode GetMeta(this ElementNode node)
+        public static ElementNode GetMeta(this ITypedElement node)
         {
             return node?.Children("meta").CastElementNodes().FirstOrDefault();
         }
 
+        public static bool IsAgeDecimalNode(this ElementNode node)
```

Comment:
Should we keep interface consistent? (ITypedElement)
Maybe we can cast to elementNode inside the function.
Or separate these functions into two extensions (ElementNodeExtension and TypedElementExtension)

Reply:
Sure. The callers of `IsAgeDecimalNode` and `IsReferenceStringNode` are `ElementNode` already, so we don't need to cast them to `ITypedElement` and then cast them back in extensions. I think using separate files would be a better idea.